src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -259,7 +264,15 @@ def _normalize_properties(self, user, subscriptions, is_service_principal, cert_
                 _IS_DEFAULT_SUBSCRIPTION: False,
                 _TENANT_ID: s.tenant_id,
                 _ENVIRONMENT_NAME: self.cli_ctx.cloud.name
-            })
+            }
+            # for Subscriptions - List REST API 2019-06-01's subscription account
+            if subscription_dict[_SUBSCRIPTION_NAME] != _TENANT_LEVEL_ACCOUNT_NAME:
```

Comment:
What about using supported_api_version() to check here?

Reply:
This is still under debate. For me, this approach looks faster and safer and is not built on any assumption about API version and the actual response body.